### PR TITLE
[DataGrid] refactor(DataGridPremium): improve grid excel export interface

### PR DESCRIPTION
**Pull request by TiagoPortfolio**
When specifying `excelOptions` -> `columnStyles` of the data grid's toolbar, I get the following error:
Screenshot 2023-05-26 at 16 34 12

I believe not all properties are required so in my change I used `Partial` to improve this.

### CODE REVIEW

**Comment by cherniavskii** [comment]
@TiagoPortfolio thank you for this pull request! Your changes seem to impact the commercially licensed code. For any changes of this nature, we require contributors to sign the MUI’s Contributor License Agreement (CLA). However, I can’t find a CLA signed that could cover these changes. Please follow the steps at https://mui-org.notion.site/CLA-Contributor-License-Agreement-92ece655b1584b10b00e4de9e67eedb0 to sign the document.

**Comment by gzrae** [comment]
CLA signed, thanks.

**Review by alexfauquette — APPROVED** [review]

**Comment by TiagoPortfolio** [comment]
Hi @alexfauquette !
I thought the PR was going to be merged automatically on close, I am going to reopen it.

**Comment by cherniavskii** [comment]
@TiagoPortfolio Thank you!

File: packages/grid/x-data-grid-premium/src/hooks/features/export/gridExcelExportInterface.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export interface GridExceljsProcessInput {
 }
 
 export interface ColumnsStylesInterface {
```
Comment (oliviertassinari) [inline packages/grid/x-data-grid-premium/src/hooks/features/export/gridExcelExportInterface.ts:18]: Off-topic. We missed the Grid prefix export on this one

```suggestion
export interface GridColumnsStylesInterface {
````

**Review by oliviertassinari — COMMENTED** [review]
I wonder if it would make sense to have a test in https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid-premium/src/tests/DataGridPremium.spec.tsx considering these are external types we don't fully control and might break between versions.

**Pull request merged**